### PR TITLE
Fixed platform settings not using unique values for Android and iOS

### LINKED ISSUE #6604: Edit Platform Settings dialog opens dirty, requiring users to save changes to close the dialog, even if they haven't made any
**Describe the bug**
Edit Platform Settings dialog opens dirty, requiring users to save changes to close the dialog, even if they haven't made any. The ideal solution would be for the dialog to close without issue UNLESS the user has actually made changes to the dialog.

**Steps to reproduce**
1. Launch the Editor.
2. Click File > Edit Platform Settings
3. Close the dialog.

**Expected behavior**
Dialog closes without issue if no changes have been made.

**Actual behavior**
Dialog requires saving changes, even if the user hasn't made any.

**Found in Branch**
development@24a3014e0

**Notes**
1. This causes issues when using autotest_mode in automated Python tests, as the Edit Platform Settings dialog will attempt to be dismissed on open, but the save dialog is also automatically dismissed, and can't be interacted with.

### PR DESCRIPTION
**Pull request by jjjoness**
Signed-off-by: John Jones-Steele <82226755+jjjoness@users.noreply.github.com>
Fixes #6604
After checking with pratikpa, Android and iOS should have unique values not shared with PC. This caused a bug when checking the dialog values against the saved values and finding them different. Now passes the AutomatedTest.

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_FileMenuOptions.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def Menus_FileMenuOptions_Work():
         ("Save As",),
         ("Save Level Statistics",),
         ("Edit Project Settings",),
-        #("Edit Platform Settings",),   Temporarily disabled due to https://github.com/o3de/o3de/issues/6604
+        ("Edit Platform Settings",),   #Temporarily disabled due to https://github.com/o3de/o3de/issues/6604
```
Comment (cgalvan) [inline AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_FileMenuOptions.py:41]: If this is fixed now, then you can actually remove that comment

**Review by cgalvan — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_FileMenuOptions.py (hunk 1 above)
Comment (jjjoness) [reply AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_FileMenuOptions.py]: I forgot to save the file before committing! Done now.

File: Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp [hunk 2]
```diff
@@ -209,12 +209,10 @@ namespace ProjectSettingsTool
                         ->Attribute(Attributes::FuncValidator, ConvertFunctorToVoid(&Validators::PackageName))
                         ->Attribute(Attributes::LinkOptional, true)
                         ->Attribute(Attributes::PropertyIdentfier, Identfiers::AndroidPackageName)
-                        ->Attribute(Attributes::LinkedProperty, Identfiers::IosBundleIdentifer)
```
Comment (AMZN-ScottR) [inline Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp:212]: FYI: if we are removing the link between these properties, the whole linked property system inside the Project Settings Tool can be removed as well.

File: Code/Editor/Plugins/ProjectSettingsTool/Validators.h [hunk 3]
```diff
@@ -24,6 +24,8 @@ namespace ProjectSettingsTool
 
         // Returns true if valid cross platform file or directory name
         FunctorValidator::ReturnType FileName(const QString& name);
+        // Returns true if valid iOS file or directory name
+        FunctorValidator::ReturnType FileNameWithPeriod(const QString& name);
```
Comment (spham-amzn) [inline Code/Editor/Plugins/ProjectSettingsTool/Validators.h:28]: If this function was meant to validate iOS filenames, then it should be named something more specific
```suggestion
        FunctorValidator::ReturnType IOSFileName(const QString& name);
```

File: Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp (hunk 2 above)
Comment (jjjoness) [reply Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp:212]: \yes, of course. Done.

File: Code/Editor/Plugins/ProjectSettingsTool/Validators.h (hunk 3 above)
Comment (jjjoness) [reply Code/Editor/Plugins/ProjectSettingsTool/Validators.h:28]: Done.

**Review by cgalvan — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**